qiskit_nature/second_q/mappers/spin_mapper.py

```diff
@@ -34,3 +37,46 @@ def map(self, second_q_op: SpinOp) -> PauliSumOp:
             The `PauliSumOp` corresponding to the problem-Hamiltonian in the qubit space.
         """
         raise NotImplementedError()
+
+    def map_all(
```

Comment:
Given that `map_all` is implemented on the base class, we should not need to duplicate it just for the type hint. There must be a cleaner solution to this. I will need to think about this longer than I have time today but I am happy to discuss :+1:

Reply:
We can use the fact that the `SpinOp` and `VibrationalOp` class are based on the `SparseLabelOp` class and only keep the `map()` method in the `QubitMapper`. I think this should avoid the duplicates